lib/Zonemaster/Backend/DB.pm

```diff
@@ -619,10 +620,10 @@ sub process_dead_test {
 sub _normalize_domain {
     my ( $domain ) = @_;
 
-    $domain = lc( $domain );
-    $domain =~ s/\.$// unless $domain eq '.';
+    # Discard errors, at this point errors have already been checked
```

Comment:
Maybe we should be so bold as to `die` if we get normalization errors at this point?

Reply:
Or maybe introduce a function in `Zonemaster::Engine::Normalization` that attempts normalization and just returns the normalized name on success instead of a list.

Reply:
I chose to die if there is an error.